lib/fennec/udp/worker.ex

```diff
@@ -38,20 +42,41 @@ defmodule Fennec.UDP.Worker do
 
   def init([dispatcher, socket, ip, port]) do
     _ = Dispatcher.register_worker(dispatcher, self(), ip, port)
-    {:ok, %{socket: socket, ip: ip, port: port}}
+    {:ok, %{socket: socket, ip: ip, port: port, turn: %TURN{}}}
   end
 
   def handle_cast({:process_data, data}, state) do
-    case Fennec.STUN.process_message!(data, state.ip, state.port) do
-      :void ->
-        :ok
-      resp ->
-        :ok = :gen_udp.send(state.socket, state.ip, state.port, resp)
-    end
-    {:noreply, state, @timeout}
+    next_state =
+      case Fennec.STUN.process_message(data, state.ip, state.port, state.turn) do
```

Comment:
Again, If I understand correctly, `TURN` *is* `STUN` on protocol level. Therefore module named `STUN` is just more generic. And it should be since we need to handle both `STUN` and `TURN` messages.